src/loaders/svg/tvgSvgLoader.cpp

```diff
@@ -2200,9 +2200,14 @@ static void _svgLoaderParserXmlOpen(SvgLoaderData* loader, const char* content,
         }
 
         if (node->type == SvgNodeType::Defs) {
-            loader->doc->node.doc.defs = node;
-            loader->def = node;
-            if (!empty) loader->stack.push(node);
+            if (loader->def && loader->doc->node.doc.defs) {
+                free(node->style);
```

Comment:
Good point :)
but.. this node is pushed to parent in createNode. (probably parent is doc)
I don't think it's good to free nodes that have already been pushed into parent's child array.

I think it's good to check if defs is already declared when doing createDefsNode().

How about this?
```cpp
--- a/src/loaders/svg/tvgSvgLoader.cpp
+++ b/src/loaders/svg/tvgSvgLoader.cpp
@@ -1066,8 +1066,9 @@ static SvgNode* _createNode(SvgNode* parent, SvgNodeType type)
 }
 
 
-static SvgNode* _createDefsNode(TVG_UNUSED SvgLoaderData* loader, TVG_UNUSED SvgNode* parent, const char* buf, unsigned bufLength)
+static SvgNode* _createDefsNode(SvgLoaderData* loader, TVG_UNUSED SvgNode* parent, const char* buf, unsigned bufLength)
 {
+    if (loader->def && loader->doc->node.doc.defs) return nullptr;
     SvgNode* node = _createNode(nullptr, SvgNodeType::Defs);
     if (!node) return nullptr;
     simpleXmlParseAttributes(buf, bufLength, nullptr, node);
@@ -2197,6 +2198,7 @@ static void _svgLoaderParserXmlOpen(SvgLoaderData* loader, const char* content,
             if (loader->stack.count > 0) parent = loader->stack.data[loader->stack.count - 1];
             else parent = loader->doc;
             node = method(loader, parent, attrs, attrsLength);
+            if (!node) return;
         }
 
         if (node->type == SvgNodeType::Defs) {
```

Reply:
Great idea!
I didn't cared about parent, as for _createDefsNode it is called _createNode(/*parent=*/ nullptr, /*type=*/ SvgNodeType::Defs); so parent is null and wont be pushed.
I think its great to abandon node creating instead of deleting it after creation. I will push a patch exactly like your idea.